slsim/Util/agn_util.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+from astropy import constants as const
+
+
+def spin_to_isco(spin):
+    """Converts dimensionless spin parameter of a black hole to the innermost stable
+    circular orbit (in gravitational radii) [R_g = GM/c^2]
+
+    :param spin: Dimensionless spin of black hole, ranging from -1 to 1.
+    :return: value of innermost stable circular orbit, ranging from 1 to 9.
+    """
+    if abs(spin) > 1:
+        raise ValueError("Absolute value of spin cannot exceed 1")
+    # Calculate intermediate values
+    z1 = 1 + (1 - spin**2) ** (1 / 3) * (
+        (1 + spin) ** (1 / 3) + (1 - spin) ** (1 / 3)
+    )
+    z2 = (3 * spin**2 + z1**2) ** (1 / 2)
+
+    # Return ISCO distance in gravitational radii
+    return 3 + z2 - np.sign(spin) * ((3 - z1) * (3 + z1 + 2 * z2)) ** (1 / 2)
+
+
+def calculate_eddington_luminosity(mass_bh_exponent):
+    """Calculates the Eddington luminosity for a black hole mass exponent.
+
+    Eddington_luminosity = 4 * pi * G * mass_bh * mass_proton
+                              * c / sigma_thompson
+
+    :param mass_bh_exponent: solution to log_10(mass_bh / mass_sun).
```

Comment:
Can you make this as a function of black hole mass and it does log10(mass_bh/mass_sun) internally? or maybe you can state this parameter as log of blackhole mass normalized by mass of the sun.

Reply:
log10(mass_bh/mass_sun) is something I do for ease of use and convenience. Typical values of log10(mass_bh/mass_sun) range from 6.0 to 10.0, are readable and do not run into any issues depending on what units are input. These in mks units are ~ 2 x 10^36 - 2 x 10^40 kg, and in cgs units are ~ 2 x 10^39 - 2 x 10^43 g.  

For units of black hole mass this is 10^6 - 10^8, and I still need to worry about someone tacking on const.M_sun as the units.

I will go with whatever you think is clearest as a user though! As long as I know what checks to put in everything will work out in the end. For now I will adjust the documentation!

Reply:
I am also changing (bh) to (black_hole) to make it clear